lib/serverengine/process_manager.rb

```diff
@@ -146,9 +151,19 @@ def spawn(*args)
       rpipe, wpipe = new_pipe_pair
 
       begin
-        options[[wpipe.fileno]] = wpipe
-        if @enable_heartbeat
-          env['SERVERENGINE_HEARTBEAT_PIPE'] = wpipe.fileno.to_s
+
+        if ServerEngine.windows?
+          @heartbeat_num += 1
+          pipe_name = "SERVERENGINE_HEARTBEAT_PIPE_" + @heartbeat_num.to_s
+          rpipe = Pipe::Server.new(pipe_name, 0, Pipe::ACCESS_DUPLEX | Pipe::FILE_FLAG_OVERLAPPED)
```

Comment:
How `fork` work on windows, by the way? If it doesn't work. we can raise an exception at `ProcessManager#fork` if windows.